src/Match.java

```diff
@@ -0,0 +1,137 @@
+import javafx.util.Pair;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Match {
+
+    enum Innings{
+        FirstInning,
+        SecondInning
+    }
+    public Team playMatch(Team A, Team B){
+        System.out.println("Match Starts");
+        Scanner sc = new Scanner(System.in);
+        int totalBalls = sc.nextInt();
+        A.setTotalBalls(totalBalls);
+        B.setTotalBalls(totalBalls);
+        if(Toss(A, B)){
+            Inning(A, B, Innings.FirstInning);
+            showScoreboard(A, B);
+            Inning(B, A, Innings.SecondInning);
+            showScoreboard(B, A);
+        }else{
+            Inning(B, A, Innings.FirstInning);
+            showScoreboard(B, A);
+            Inning(A, B, Innings.SecondInning);
+            showScoreboard(A, B);
+        }
+        if(A.getRuns()>B.getRuns()) return A;
+        else if(A.getRuns()<B.getRuns()) return B;
+        return null;
+    }
+
+    private static void Inning(Team A, Team B, Innings inning){
+        Player Striker = A.getNextBatsman();
+        Player nonStriker = A.getNextBatsman();
+        Player bowler = B.getNextBowler();
+        int totalBalls = A.getTotalBalls();
+        int totalOvers = totalBalls/6;
+        for (int i = 1; i <= totalOvers; i++) {
+            for (int j = 0; j < 6; j++) {
+                int outcome = ballOutcome();
+                A.incBallsPlayed(Striker);
+                bowler.incBallsBowled();
+                if(outcome==7){
+                    A.incWickets(Striker, bowler);
+                    if(A.getWickets()<10)
+                        Striker = A.getNextBatsman();
+                }else {
+                    A.incRuns(outcome);
+                    Striker.incRunsScored(outcome);
+                    bowler.incRunsConceeded(outcome);
+                    if(outcome%2==1){
+                        Player temp = Striker;
+                        Striker = nonStriker;
+                        nonStriker = temp;
+                    }
+                }
+                if(A.getWickets()==10 || (inning.compareTo(Innings.SecondInning)==0 && A.getRuns()>B.getRuns()))
+                    break;
+            }
+            Player temp = Striker;
+            Striker = nonStriker;
+            nonStriker = temp;
+            if(A.getWickets()==10 || (inning.compareTo(Innings.SecondInning)==0 && A.getRuns()>B.getRuns()))
+                break;
+            Player prevBowler = bowler;
+            bowler = B.getNextBowler();
+            if(prevBowler.getBallsBowled()== totalBalls/5 )prevBowler.setBowlerStatus(statusOfBowler.ReachedMaxOverLimit);;
+            prevBowler.setBowlerStatus(statusOfBowler.CanBowlNext);
+        }
+    }
+    private static int ballOutcome(){
+        int outcome = (int)(Math.random() * 72);;
+        return outcome/10;
+    }
+
+    public static boolean Toss(Team A, Team B){
```

Comment:
create some helper class and move some of the codes there and call it functionally from here